go/service/chat_test.go

```diff
@@ -0,0 +1,112 @@
+package service
+
+// Copyright 2016 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+import (
+	"testing"
+
+	"github.com/keybase/client/go/libkb"
+	keybase1 "github.com/keybase/client/go/protocol"
+	"github.com/keybase/gregor/protocol/chat1"
+)
+
+func cryptKey(t *testing.T) *keybase1.CryptKey {
+	kp, err := libkb.GenerateNaclDHKeyPair()
+	if err != nil {
+		t.Fatal(err)
+	}
+	return &keybase1.CryptKey{
+		KeyGeneration: 1,
+		Key:           keybase1.Bytes32(*kp.Private),
+	}
+}
+
+func textMsg(text string) keybase1.MessagePlaintext {
+	return keybase1.MessagePlaintext{
+		ClientHeader: chat1.MessageClientHeader{},
+		MessageBodies: []keybase1.MessageBody{
```

Comment:
Hmm.. looks like MessageType could be removed from MessageClientHeader and this wouldn't be a problem.